pkg/admission/validator/shoot_validator.go

```diff
@@ -30,66 +35,152 @@ type shoot struct {
+		return fmt.Errorf("gardener rules not set")
+	}
+	// RulesRef will be set to default val as not a pointer
+	return nil
+}
+
+func verifyWebhook(falcoConf *service.FalcoServiceConfig) error {
+	webhook := falcoConf.CustomWebhook
+	if webhook == nil {
+		return fmt.Errorf("webhook is nil")
+	} else if webhook.Enabled == nil {
+		return fmt.Errorf("webhook needs to be either enabled or disbaled")
+	} else if *webhook.Enabled && webhook.Address == nil {
+		return fmt.Errorf("webhook is enabled but without address")
+	}
+	// may also want to enforce headers at some point
+	return nil
+}
+
+func verifyResources(falcoConf *service.FalcoServiceConfig) error {
+	resource := falcoConf.Resources
+	if resource == nil {
+		return fmt.Errorf("resource is not defined")
+	}
+	if *resource != "gardener" && *resource != "falcoctl" {
+		return fmt.Errorf("resource needs to be either gardener or falcoctl")
+	}
+	return nil
+}
+
+func verifyFalcoVersion(falcoConf *service.FalcoServiceConfig) error {
+	if err := verifyFalcoVersionInVersions(falcoConf, falco.FalcoVersions().Falco); err != nil {
+		return err
+	}
+	return nil
+}
+
+func verifyFalcoVersionInVersions(falcoConf *service.FalcoServiceConfig, versions *falcoversions.FalcoVersions) error {
+	chosenVersion := falcoConf.FalcoVersion
+	if chosenVersion == nil {
+		return fmt.Errorf("falcoVersion is nil")
+	}
+
+	for _, ver := range versions.FalcoVersions {
+		if *chosenVersion == ver.Version {
+			if ver.Classification == "deprecated" {
+				return fmt.Errorf("chosen version is marked as deprecated")
+			}
+			return nil
+		}
+	}
+	return fmt.Errorf("version not found in possible versions")
 }
 
 // isDisabled returns true if extension is explicitly disabled.
 func (s *shoot) isDisabled(shoot *core.Shoot) bool {
 	ext := s.findExtension(shoot)
 	if ext == nil {
-		return false
+		return true
 	}
+
 	if ext.Disabled != nil {
 		return *ext.Disabled
 	}
 	return false
 }
 
-// extractDNSConfig extracts DNSConfig from providerConfig.
-func (s *shoot) extractDNSConfig(shoot *core.Shoot) (*apisservice.DNSConfig, error) {
-	ext := s.findExtension(shoot)
-	if ext != nil && ext.ProviderConfig != nil {
-		dnsConfig := &apisservice.DNSConfig{}
-		if _, _, err := s.decoder.Decode(ext.ProviderConfig.Raw, nil, dnsConfig); err != nil {
-			return nil, fmt.Errorf("failed to decode %s provider config: %w", ext.Type, err)
-		}
-		return dnsConfig, nil
-	}
-
-	return nil, nil
-}
-
-// findExtension returns shoot-dns-service extension.
+// findExtension returns shoot-falco-service extension.
 func (s *shoot) findExtension(shoot *core.Shoot) *core.Extension {
+	extensionType := "shoot-falco-service"
 	for i, ext := range shoot.Spec.Extensions {
-		if ext.Type == service.ExtensionType {
+		if ext.Type == extensionType {
 			return &shoot.Spec.Extensions[i]
 		}
 	}
 	return nil
 }
-*/
+
+func (s *shoot) extractFalcoConfig(shoot *core.Shoot) (*service.FalcoServiceConfig, error) {
```

Comment:
Duplicate, see above

Reply:
Needs to stay as shoot is different struct for validator and mutator